Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -459,12 +461,14 @@ void SourceBufferPrivate::computeEvictionData(ComputeEvictionDataRule rule)
             return 0;
         size_t evictableSize = 0;
         auto currentTime = mediaSource->currentTime();
+
+        // We can evict everything from the beginning of the buffer to a maximum of timeChunk (3s) before currentTime (or the previous sync sample whichever comes first).
         auto timeChunkAsMilliseconds = evictionAlgorithmTimeChunkLowThreshold;
         const auto timeChunk = MediaTime(timeChunkAsMilliseconds, 1000);
         const auto maximumRangeEndBeforeCurrentTime = std::min(currentTime - timeChunk, findPreviousSyncSamplePresentationTime(currentTime));
 
         auto rangeStartBeforeCurrentTime = minimumBufferedTime();
-        auto rangeEndBeforeCurrentTime = std::min(rangeStartBeforeCurrentTime, maximumRangeEndBeforeCurrentTime);
+        auto rangeEndBeforeCurrentTime = maximumRangeEndBeforeCurrentTime;
```

Comment:
Why have both rangeEndBeforeCurrentTime and maximumRangeEndBeforeCurrentTime?
Can we just have maximumRangeEndBeforeCurrentTime?

Reply:
I had tried to stay as close to the code actually performing the eviction.